### PR TITLE
feat: post ID less-frontrunnable by depending on author's sequential ID instead of global one

### PR DESCRIPTION
**Pull request by vicnaum**
Link T-23023

### CODE REVIEW

**Comment by vicnaum** [comment]
Link T-23023

File: contracts/core/primitives/feed/FeedCore.sol [hunk 1]
```diff
@@ -7,7 +7,7 @@ import "./../../libraries/ExtraDataLib.sol";
 
 struct PostStorage {
     address author;
-    uint256 localSequentialId;
+    uint256 localSequentialId; // TODO: should we accent that this is now the author-based sequential ID?
```
Comment (donosonaumczuk) [inline contracts/core/primitives/feed/FeedCore.sol]: No, I think we should have both! `localSequentialId`, `authorLocalSequentialId` or similar

File: contracts/core/primitives/feed/Feed.sol [hunk 2]
```diff
@@ -270,4 +270,8 @@ contract Feed is IFeed, RuleBasedFeed, AccessControlled {
     function getExtraData(bytes32 key) external view override returns (bytes memory) {
         return Core.$storage().extraData[key];
     }
+
+    function getAuthorPostId(address author, uint256 localSequentialId) external view returns (uint256) {
```
Comment (donosonaumczuk) [inline contracts/core/primitives/feed/Feed.sol]: I think we should have the two getters, `getPostLocalSequentialId` and `getPostAuthorLocalSequentialId`

File: contracts/core/primitives/feed/FeedCore.sol [hunk 3]
```diff
@@ -72,16 +73,17 @@ library FeedCore {
         return $storage().extraData.set(extraDataToSet);
     }
 
-    function _generatePostId(uint256 localSequentialId) internal view returns (uint256) {
-        return uint256(keccak256(abi.encode("evm:", block.chainid, address(this), localSequentialId)));
+    function _generatePostId(address author, uint256 localSequentialId) internal view returns (uint256) {
+        return uint256(keccak256(abi.encode("evm:", block.chainid, address(this), author, localSequentialId)));
```
Comment (donosonaumczuk) [inline contracts/core/primitives/feed/FeedCore.sol]: Here we could maybe use both sequential IDs, not sure. I wonder if there is any advantage for this

**Review by donosonaumczuk — CHANGES_REQUESTED** [review]
Conclusion: I think we should keep both sequential IDs, the "global" for the feed is super useful, like "this is the Xth post of the feed"

File: contracts/core/primitives/feed/FeedCore.sol (hunk 3 above)
Comment (vicnaum) [reply contracts/core/primitives/feed/FeedCore.sol]: We cannot, cause someone could frontrun the primitive localSequentialId

Comment (donosonaumczuk) [reply contracts/core/primitives/feed/FeedCore.sol]: Ah true, the whole idea was to make it deterministic haha. Yeah, then this is perfect like it is

File: contracts/core/primitives/feed/Feed.sol [hunk 4]
```diff
@@ -270,4 +270,8 @@ contract Feed is IFeed, RuleBasedFeed, AccessControlled {
...
+        return Core._generatePostId(author, localSequentialId);
```
Comment (donosonaumczuk) [inline contracts/core/primitives/feed/Feed.sol]: Actually, this is wrong, this is the Post ID, not the author local sequential one

**Review by donosonaumczuk — CHANGES_REQUESTED** [review]

File: contracts/core/primitives/feed/FeedCore.sol (hunk 1 above)
Comment (donosonaumczuk) [reply contracts/core/primitives/feed/FeedCore.sol]: Added them

File: contracts/core/primitives/feed/Feed.sol (hunk 2 above)
Comment (donosonaumczuk) [reply contracts/core/primitives/feed/Feed.sol]: Added both of them

File: contracts/core/primitives/feed/Feed.sol (hunk 4 above)
Comment (donosonaumczuk) [reply contracts/core/primitives/feed/Feed.sol]: Not sure what this getter was

**Pull request merged**